src/engine/BoundManager.h

```diff
@@ -0,0 +1,128 @@
+ **   * tighten*Bounds - shared method to update bounds, propagates the new bounds
+ **                        to the _tableau (if registered) to keep the assignment and
+ **                      basic/non-basic variables updated accordingly.
+ **
+ ** As soon as bounds become inconsistent, i.e. lowerBound > upperBound, an
+ ** InfeasableQueryException is thrown. In the long run, we want the exception
+ ** replaced by a flag, and switch to the conflict analysis mode instead.
+ **
+ ** It is assumed that variables are not introduced on the fly, and as such
+ ** interaction with context-dependent features is not implemented.
+ **/
+
+#ifndef __BoundManager_h__
+#define __BoundManager_h__
+
+#include "context/cdo.h"
+#include "context/context.h"
+#include "List.h"
+#include "Vector.h"
+
+class Tableau;
+class Tightening;
+class BoundManager
+{
+public:
+    BoundManager( CVC4::context::Context &ctx );
+    ~BoundManager();
+
+    /*
+       Initialize BoundManager and register numberOfVariables of variables
+     */
+    void initialize( unsigned numberOfVariables );
+
+    /*
+       Registers a new variable, grows the BoundManager size and bound vectors,
+       initializes new bounds to +/-inf, and returns the index of the new
+       variable.
+     */
+    unsigned registerNewVariable();
+
+    /*
+       Returns number of registered variables
+     */
+    unsigned getNumberOfVariables();
+
+    /*
+       Communicates bounds to the bound Manager and informs _tableau of the
+       changes, so that any necessary updates can be performed.
+     */
+    bool tightenLowerBound( unsigned variable, double value );
+    bool tightenUpperBound( unsigned variable, double value );
+
+    /*
+       Silently sets bounds to the assigned value, checks bound consistency.
+     */
+    bool setLowerBound( unsigned variable, double value );
+    bool setUpperBound( unsigned variable, double value );
+
+    /*
+       Return current bound value.
+     */
+    double getLowerBound( unsigned variable );
+    double getUpperBound( unsigned variable );
+
+    /*
+       Obtain a list of all the bound updates since the last call to
+       getTightenings.
+     */
+    void getTightenings( List<Tightening> &tightenings );
+
+    /*
+          Returns true if the bounds for the variable is valid, used to
+          detect a conflict state.
+    */
+    bool consistentBounds( unsigned variable );
+
+
+    /*
+       Register Tableau reference for callbacks from tighten*Bound methods. 
+     */
+    void registerTableauReference( Tableau *tableau );
+
+private:
+    CVC4::context::Context &_context;
+    unsigned _size;
+    Tableau *_tableau; // Used only by callbacks
+
+    Vector<CVC4::context::CDO<double> *> _lowerBounds;
+    Vector<CVC4::context::CDO<double> *> _upperBounds;
+
+    Vector<CVC4::context::CDO<bool> *> _tightenedLower;
+    Vector<CVC4::context::CDO<bool> *> _tightenedUpper;
+};
+
+
+#endif // __BoundManager_h__
+
+//
```

Comment:
this suffix can be removed, it became mostly irrelevant when we switched to cmake